src/TurbulenceClosures/discrete_diffusion_function.jl

```diff
@@ -5,46 +5,67 @@ using Oceananigans.Utils: instantiate
     struct DiscreteDiffusionFunction{LX, LY, LZ, P, F} 
 
 A wrapper for a diffusivity functions with optional parameters at a specified locations.
+"""
+struct DiscreteDiffusionFunction{LX, LY, LZ, P, F} 
+    func :: F
+    parameters :: P
+
+    DiscreteDiffusionFunction{LX, LY, LZ}(func::F, parameters::P) where {LX, LY, LZ, F, P} =
+        new{LX, LY, LZ, P, F}(func, parameters)
+end
 
-If LX == LY == LZ == nothing the function call requires locations in the signature
-When `parameters=nothing`, the diffusivity `func` is called with the signature
+"""
+    DiscreteDiffusionFunction(func; parameters, loc)
 
-```
-func(i, j, k, grid, lx, ly, lz, clock, model_fields)
-```
+Return a discrete representation of a diffusivity `func`tion with optional `parameters`
+at specified `loc`ations.
 
-where `i, j, k` are the indices,
-where `grid` is `model.grid`, `clock.time` is the current simulation time and
-`clock.iteration` is the current model iteration, and
-`model_fields` is a `NamedTuple` with `u, v, w`, the fields in `model.tracers` and the `model.auxiliary_fields`,
+Keyword Arguments
+=================
 
-When `parameters` is not `nothing`, `func` is called with the signature
+* `parameters`: A named tuple with parameters used by `func`; default: `nothing`.
 
-```
-func(i, j, k, grid, lx, ly, lz, clock, model_fields, parameters)
-```
+* `loc`: A tuple with the locations that the diffusivity `func` is applied on.
 
-If LX, LY, LZ != (nothing, nothing, nothing) the function call does not require locations in the signature
-and the output will be automatically interpolated on the correct location
+  **Without locations**
 
-without parameters
-```
-func(i, j, k, grid, clock, model_fields)
-```
-with parameters
-```
-func(i, j, k, grid, clock, model_fields, parameters)
-```
-"""
-struct DiscreteDiffusionFunction{LX, LY, LZ, P, F} 
-    func :: F
-    parameters :: P
+  If `LX == LY == LZ == nothing` the diffusivity is evaluated at the required locations. In this case, the function call *requires passing* locations `ℓx, ℓy, ℓz` in the signature.
+  In this case, the diffusivity `func` is called with the signature:
```

Comment:
```suggestion
  If `LX == LY == LZ == nothing` diffusivity `func` is called with the signature:
```

This confused me a little bit. Is this the default? I think it's important to concisely describe what the user input needs to be. The explanation that "the diffusivity is evaluated at the required locations" doesn't help I don't think.

Reply:
Once the user input is described, one could add a follow-up sentence like "the purpose of this feature is..."

Reply:
So there is no default for the `DiscreteDiffusionFunction`; seems like it's a required kwarg.

But various closures have defaults and I added remarks on that in those docs; see [8ce5499](https://github.com/CliMA/Oceananigans.jl/pull/3133/commits/8ce5499e2df2af29c80644b1b3e0e40e6a1b9d03).

Reply:
@glwagner, are you happy with how the docstrings of ScalarDiffusivity/ScalarBiharmonicDiffusivity explain the situation?

The `DiscreteDiffusionFunction` docs are not user-facing (they are not even included in the Docs/Library). Happy to make them better tho.

Reply:
I think like DiscreteForcing and DiscreteBoundaryFunction, DiscreteDiffusionFunction is not user-facing as @simone-silvestri says. Users build this object through ScalarDiffusivity right?

Reply:
Well, maybe I was confused. I was just looking at the edits in the PR.

Reply:
I think if its not user facing the docs should actually say that. Also the docs should be oriented towards developers, not users.

Reply:
In summary I am fine for the updates proposed on these developer-facing docs. I suspect there are some possible improvements of the user API; as the discussion reveals there are still some wrinkles

Reply:
Not all of the changes in this PR was about developer-facing docs though.